controllers/clustergroupupgrade_controller.go

```diff
@@ -1628,23 +1591,24 @@ func (r *ClusterGroupUpgradeReconciler) getAllClustersForUpgrade(ctx context.Con
 	return clusterNames, nil
 }
 
-func (r *ClusterGroupUpgradeReconciler) updateChildResourceNames(ctx context.Context, safeNameMap map[string]string, childResourceNameList []string, u *unstructured.Unstructured) ([]string, error) {
-	if name, ok := u.GetAnnotations()[utils.ResourceNameAnnotation]; ok {
-		if safeName, ok := safeNameMap[name]; ok {
+// checkDuplicateChildResources goes through the list of resources against the current names map. Duplicate (with same desired name annotation value) ones are removed.
```

Comment:
Please check the format for other function descriptions used throughout the file, for ex getNextNonCompliantPolicyForCluster

Reply:
Done